frontend/src/pages/EditRolesPage/components/buttons/EditRolesButton.tsx

```diff
@@ -0,0 +1,150 @@
+import React, { Profiler } from 'react';
+import { useHistory } from 'react-router-dom';
+
+import { Button } from '@SharedComponents';
+import * as ROUTES from '@Constants/routes';
+import {
+  AppUserResponse,
+  AppUserInterviewAvailability,
+  InterviewDatesResponse,
+} from '@Services/api';
+import {
+  CssBaseline,
+  Drawer,
+  AppBar,
+  Toolbar,
+  List,
+  ListItem,
+  ListItemIcon,
+  ListItemText,
+  Typography,
+  Divider,
+  IconButton,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogContentText,
+  DialogTitle,
+  Hidden,
+} from '@material-ui/core';
+import { updateUserById } from '@Services/UserService';
+import { AppUserPostRequest } from '@Services/api';
+
+interface EditRolesButtonProps {
+  id: number;
+  role: string;
+  email: string;
+  firstName: string;
+  lastName: string;
+}
+interface INITIAL_STATES {
+  isPopupOpen?: boolean;
+  id?: number;
+  role?: string;
+  email?: string;
+  firstName?: string;
+  lastName?: string;
+}
+
+class EditRolesButton extends React.Component<
+  EditRolesButtonProps,
+  INITIAL_STATES
+> {
+  constructor(props: EditRolesButtonProps) {
+    super(props);
+
+    this.state = {
+      isPopupOpen: false,
+      id: props.id,
+      role: props.role,
+      email: props.email,
+      firstName: props.firstName,
+      lastName: props.lastName,
+    };
+    this.handleEditRolePopup = this.handleEditRolePopup.bind(this);
+    this.handleEditRoleClose = this.handleEditRoleClose.bind(this);
+    this.updateUserRole = this.updateUserRole.bind(this);
+  }
+
+  handleEditRolePopup() {
+    this.setState({
+      isPopupOpen: true,
+    });
+  }
+  handleEditRoleClose() {
+    this.setState({
+      isPopupOpen: false,
+    });
+  }
+  async updateUserRole(newRole) {
+    const updatedUser: AppUserPostRequest = {
+      role: newRole,
+      email: this.state.email!,
+      firstName: this.state.firstName!,
+      lastName: this.state.lastName!,
+    };
+    try {
+      await updateUserById(this.state.id!, updatedUser);
+      this.setState({
+        role: newRole,
+      });
+    } catch (error) {}
+    window.location.reload();
```

Comment:
we shouldn't be refreshing the window from a react component, what are we trying to do here?

Reply:
This was a temporary fix for testing. I was trying to figure out a way to make it so the edited roles we submit upon popup show up on the table, since the rows themselves are only fetched once instead of updated automatically. I'm still trying to figure out a solution for this, so if you have any suggestions I'd greatly appreciate them. The rows aren't state variables either which makes it a bit harder for me to figure out.

Reply:
`EditRolesButton` can take in a callback from the parent component that sets the state of the parent component